packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -214,11 +215,53 @@ class WebReleaseBundle extends Target {
     }
     final Directory outputDirectory = environment.outputDir.childDirectory('assets');
     outputDirectory.createSync(recursive: true);
-    environment.projectDir
-      .childDirectory('web')
-      .childFile('index.html')
-      .copySync(fs.path.join(environment.outputDir.path, 'index.html'));
-    final Depfile depfile = await copyAssets(environment, environment.outputDir.childDirectory('assets'));
-    depfile.writeToFile(environment.buildDir.childFile('flutter_assets.d'));
+    final String basePath = environment.projectDir.childDirectory('web').absolute.path;
+    final List<File> inputs = <File>[];
+    final List<File> outputs = <File>[];
+    final String outputPrefix = fs.path.join(
+      environment.projectDir.path,
+    );
+    for (String artifact in environment.projectDir.childDirectory('web').listSync(recursive: true).map((FileSystemEntity file) => file.absolute.path)) {
```

Comment:
recursive: true will give you nested files with the correct path. you could probably update this to:

`final List<File> files = environment.projectDir.childDirectory('web').listSync(recursive: true).whereType<File>().toList();`

This functionality could also be pulled out into a helper function to make it easier to read